PR for review regarding changes to upstream

Licenser opened the pull request:
This PR is to review changes the Ky/community/hex version of cuttlefish has and prepare for a 'clean' and approved merge to upstream cuttlefish.

tsloughter committed:
update for rebar3

tsloughter committed:
move neotoma to dev profile

tsloughter committed:
add hex metadata

tsloughter committed:
set namespace on escriptize provider hook

tsloughter committed:
upgrade lager for OTP 19

larshesel committed:
Make it compile on Erlang/OTP 20

Ignore the new warning introduced for modules using `export_all`.

tsloughter committed:
Merge pull request #1 from larshesel/compile-on-erlang-otp-20

Make it compile on Erlang/OTP 20

larshesel committed:
Make escript work on OTP20

The `-smp` flag has been removed on OTP 20.

tsloughter committed:
Merge pull request #2 from larshesel/erlang-otp-20-support-take-2

Make escript work on OTP20

Licenser commented on rebar.config:
```diff
@@ -1,23 +1,20 @@
 
-{eunit_opts, [verbose]}.
-{cover_enabled, true}.
+{deps, [getopt, lager]}.
```
Versioned dependencies would be nice here to make it clear which versions we require and prevent accidental upgrades.

Licenser commented on .gitignore:
```diff
@@ -1,15 +1,15 @@
 generated.conf
 ebin/*
 .eunit
-src/conf_parse.erl
```
Why do we commit a generated file to the repo?

Licenser commented on src/cuttlefish.app.src:
```diff
@@ -7,5 +7,9 @@
+
+  {maintainers,[]},
+  {licenses,["Apache"]},
+  {links,[{"Github","https://github.com/tsloughter/cuttlefish"}]}
```
have this point here or to the basho repo so people find the right repo when clicking on the link in hex?

Licenser commented on src/cuttlefish_duration_parse.erl:
```diff
@@ -0,0 +1,338 @@
+-module(cuttlefish_duration_parse).
```
why commit a generated file?

Licenser commented on src/cuttlefish_unit.erl:
```diff
@@ -1,7 +1,7 @@
 
 -include_lib("eunit/include/eunit.hrl").
--compile(export_all).
+-compile([nowarn_export_all, export_all]).
```
shouldn't this live under `test` if it is a unit test file?

Commit:
prep for test boot

Commit:
continuing on mix adventure

Commit:
escript runs - removed parse transform targets in cuttlefish.erl

Commit:
remove inspirations

Commit:
Update mix.exs

Commit:
removed parse_transform from rebar config

Commit:
suppress compiler warning (fill in package details later)

Commit:
Merge branch 'develop' of github.com:bryanhuntesl/cuttlefish into develop

Licenser committed:
Merge pull request #5 from bryanhuntesl/develop

make cuttlefish work with erlang 20 / elixir 1.5.2

ghost commented:
Any indication of timescale when this will get merged upstream? Or is it the case this will become the canonical repository?

Licenser commented:
I'm not sure. I've had zero luck so far with any upstream merges even for simpler libraries with fewer changes. (perhaps no one saw them just like I missed this PR :P) My personal preference would be to have one canonical place for most if not all things just so people don't get confused.

`riak_core_ng` might be a special case as I'm unsure if it is good to merge upstream of suposed to be merged upstream or wanted to be merged upstream, but the whole set of libs would be nice to reduce the number of forks.

Licenser committed:
Merge branch 'upstream' into develop

Commit:
Let's not lock it down to specific Elxir version

More hassle than it's worth, also currently 0 Elixir projects using it. Let's have the minimum constraints. Should never have specified the Elixir version.

Licenser committed:
Merge pull request #6 from bryanhuntesl/patch-1

Let's not lock it down to specific Elixir version

Licenser committed:
Revert "Let's not lock it down to specific Elixir version"

Licenser committed:
Merge pull request #8 from Kyorai/revert-6-patch-1

Revert "Let's not lock it down to specific Elixir version"

Licenser committed:
Revert "make cuttlefish work with erlang 20 / elixir 1.5.2"

Licenser committed:
Merge pull request #9 from Kyorai/revert-5-develop

Revert "make cuttlefish work with erlang 20 / elixir 1.5.2"

Licenser committed:
Add hashes

Licenser committed:
Bump lager

dumbbell committed:
Don't warn about erlang:get_stacktrace/0 deprecation

dumbbell committed:
Rmeove -compile(export_all)

eunit exports testcase automatically.

dumbbell committed:
Support bbmustache to render test templates

There is no Mustache implementation in Rebar 3 apparently, unlike Rebar
2.

dumbbell committed:
Use unicode:characters_to_binary/1 to convert an UTF-8 string

dumbbell committed:
Paths in tests are relative to project root dir

The current working directory in the context of test must have changed
in Rebar 3 (?).

dumbbell committed:
Enable Travis CI + export test coverage

dumbbell committed:
Fix #translation{} definition: `func` can be undefined

The error was reported by Dialyzer.

dumbbell committed:
Support reading files & schemas from .ez archives

This is useful when Erlang applications are packaged as .ez archives:
this way, Cuttlefish can read their configuration schema directly. The
caller does not need to extract the file(s) and copy them somewhere
else.

Licenser committed:
Merge pull request #10 from rabbitmq/fix-tests

Fix tests + enable CI

Licenser committed:
README.md: Add CI & Hex.pm badges

(cherry picked from commit c8687d9c5e0e4b0ae36fec9899f48110989e5964)

michaelklishin committed:
Bump Lager to a version compatible with modern Erlang/OTP releases

michaelklishin committed:
Merge pull request #13 from Kyorai/lager-3.7.x

Bump Lager to 3.7.x

dumbbell committed:
Merge pull request #12 from rabbitmq/read-files-from-ez-archives

Support reading files & schemas from .ez archives

dumbbell committed:
Merge pull request #14 from Kyorai/integrations-take-2
